tests/bin/utils.ts

```diff
@@ -72,69 +73,97 @@ async function pkStdio(
   // (if not defined in the env) to ensure no attempted connections. A regular
   // PolykeyAgent is expected to initially connect to the mainnet seed nodes
   env['PK_SEED_NODES'] = env['PK_SEED_NODES'] ?? '';
-  // Parse the arguments of process.stdout.write and process.stderr.write
-  const parseArgs = (args) => {
-    const data = args[0];
-    if (typeof data === 'string') {
-      return data;
-    } else {
-      let encoding: BufferEncoding = 'utf8';
-      if (typeof args[1] === 'string') {
-        encoding = args[1] as BufferEncoding;
+  if (global.testCmd != null) {
+    // If using the command override we need to spawn a process
+    env = {
+      ...process.env,
+      ...env,
+    };
+    const command = path.resolve(path.join(global.projectDir, global.testCmd));
+    const subprocess = child_process.spawn(command, [...args], {
+      env,
+      cwd,
+      stdio: ['pipe', 'pipe', 'pipe'],
+      windowsHide: true,
+    });
```

Comment:
This now swaps `pkStdio` to use `spawn` in the case of targeting an external executable. This requires comments up top about how the `pkStdio` changes behaviour.